File: ruby_env/scripts/run_api.rb
#!/usr/bin/env ruby
# encoding: utf-8

# Script to run the Sinatra API server

require 'fileutils'

# ANSI color codes
COLORS = {
  green: "\033[32m",
  yellow: "\033[33m",
  red: "\033[31m",
  blue: "\033[34m",
  cyan: "\033[36m",
  magenta: "\033[35m",
  reset: "\033[0m"
}

# Colorize text
def colorize(text, color)
  "#{COLORS[color]}#{text}#{COLORS[:reset]}"
end

# Print header
puts colorize("\n" + "=" * 50, :cyan)
puts colorize("STARTING API SERVER".center(50), :cyan)
puts colorize("=" * 50 + "\n", :cyan)

# Change to project root directory
root_dir = File.expand_path('..', File.dirname(__FILE__))
Dir.chdir(root_dir)

# Create db directory if it doesn't exist
FileUtils.mkdir_p('db')

# Create public directory if it doesn't exist
FileUtils.mkdir_p('public')

# Create views directory if it doesn't exist
FileUtils.mkdir_p('views')

# Check if database migrations exist
if Dir.glob('db/migrate/*.rb').empty?
  puts colorize("No migrations found. Make sure to run migrations before starting the server.", :yellow)
  puts colorize("Run: bundle exec rake db:migrate", :blue)
end

# Run the server
puts colorize("Starting Sinatra server at http://localhost:4567", :green)
puts colorize("Press Ctrl+C to stop the server", :yellow)
puts

# Start the server
exec "bundle exec ruby app.rb -o 0.0.0.0"

File: python_env/scripts/run_api.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Script to run the FastAPI application
"""

import os
import sys
import uvicorn
from pathlib import Path

# ANSI color codes
COLORS = {
    'GREEN': '\033[32m',
    'YELLOW': '\033[33m',
    'RED': '\033[31m',
    'BLUE': '\033[34m',
    'CYAN': '\033[36m',
    'MAGENTA': '\033[35m',
    'RESET': '\033[0m'
}

def colorize(text, color):
    """
    Add color to the given text
    
    Args:
        text: Text to colorize
        color: Color to use from COLORS dict
    
    Returns:
        Colorized text string
    """
    return f"{COLORS[color]}{text}{COLORS['RESET']}"


def run_api(host="127.0.0.1", port=8000, reload=True):
    """
    Run the FastAPI application
    
    Args:
        host: Host to bind the server to
        port: Port to bind the server to
        reload: Whether to reload the server on code changes
    """
    print(colorize("\n" + "="*50, 'CYAN'))
    print(colorize("STARTING API SERVER".center(50), 'CYAN'))
    print(colorize("="*50 + "\n", 'CYAN'))
    
    # Change to project root directory
    os.chdir(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
    
    # Ensure src directory is in path
    sys.path.insert(0, os.getcwd())
    
    print(colorize(f"Starting FastAPI server at http://{host}:{port}", 'GREEN'))
    print(colorize("API documentation will be available at:", 'BLUE'))
    print(colorize(f"http://{host}:{port}/docs", 'BLUE'))
    print(colorize(f"http://{host}:{port}/redoc", 'BLUE'))
    
    # Run the server
    uvicorn.run(
        "src.app:app",
        host=host,
        port=port,
        reload=reload,
        log_level="info"
    )


if __name__ == "__main__":
    run_api()
